File: project/server/src/config/logger.ts
type LogLevel = "info" | "warn" | "error";

function format(level: LogLevel, message: string): string {
  return `[${new Date().toISOString()}] [${level.toUpperCase()}] ${message}`;
}

export const logger = {
  info(message: string): void {
    console.log(format("info", message));
  },
  warn(message: string): void {
    console.warn(format("warn", message));
  },
  error(message: string): void {
    console.error(format("error", message));
  },
};

File: project/agent/src/modules/infra/log/logger.cpp
#include "infra_log/logger.hpp"

#include <iostream>

namespace piguard::infra {

std::string Logger::name() const { return "Logger"; }

bool Logger::start() { return true; }

void Logger::stop() {}

void Logger::log(Level level, const std::string& message) const {
    static const char* kText[] = {"DEBUG", "INFO", "WARN", "ERROR", "FATAL"};
    std::cout << "[" << kText[static_cast<int>(level)] << "] " << message << '\n';
}

}  // namespace piguard::infra
